src/auth/oauth2client.ts

```diff
@@ -749,7 +754,13 @@ export class OAuth2Client extends AuthClient {
       !this.credentials.access_token || this.isTokenExpiring();
     if (shouldRefresh) {
       if (!this.credentials.refresh_token) {
-        throw new Error('No refresh token is set.');
+        const downscopedCreds = await this.refreshHandler();
```

Comment:
Let's not use `downscopedCreds` as a name as this can be used in non-downscoping scenarios.

Reply:
Also since this is an optional property `refreshHandler`. You need to check it exists before running it.